arm-software/embedded/arm-runtimes/test-support/lit-wrapper.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python3
+
+# SPDX-FileCopyrightText: Copyright 2024-2025 Arm Limited and/or its affiliates <open-source-office@arm.com>
+
+# This is a wrapper script for LLVM's lit, which takes additional options in
+# order to prepare a specific environment for running tests.
+# The new --xfails-file and --xfails-not-file options are functionally similar
+# to the existing --xfails and --xfails-not options, but allow for a test
+# list to be loaded from a file instead of on the command line. This is
+# preferable when there are large number of tests that need to be specified.
+
+import argparse
+import os
+import runpy
+import sys
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--xfails-file")
+parser.add_argument("--xfails-not-file")
+parser.add_argument("--real-lit-path", required=True)
+# Only parse the known args above. The rest should be processed the actual lit.
+known_args, unknown_args = parser.parse_known_args()
+
+# If an xfails file is set, read it.
+if known_args.xfails_file:
+    if os.path.isfile(known_args.xfails_file):
+        with open(known_args.xfails_file, "r") as f:
```

Comment:
Consider using a linter on this file. I believe pylint would for instance complain about the lack of the `encoding` argument here.